src/platforms/common/blackpill-f4/README.md

```diff
@@ -26,39 +31,45 @@ This directory contains the common code for the following platforms:
 | LED UART        | PA4    | PA1                  | LED       |
 | User button KEY | PA0    | PA0                  |           |
 
-## Instructions for Makefile buildsystem and ST MaskROM
-### How to Build
+## Instructions for build system
 
-In the following commands replace `blackpill-f4x1cx` with the platform you are using, e.g. `blackpill-f401cc`.
+### Variant with meson and BMD bootloader enabled (default)
 
-To build the code using the default pinout, run:
+0. Clone the repo and libopencm3 submodule, install toolchains, meson, etc.
 
 ```sh
+git clone https://github.com/blackmagic-debug/blackmagic.git
 cd blackmagic
-make clean
-make PROBE_HOST=blackpill-f4x1cx
+git submodule update --init
 ```
 
-or, to use alternative pinout 1, run:
+1. Create a build configuration for specific platform (WeActStudio F401CC, F401CE, or F411CE) with specific options (enable BMD bootloader flags and offsets -- this is the default)
 
 ```sh
-cd blackmagic
-make clean
-make PROBE_HOST=blackpill-f4x1cx ALTERNATIVE_PINOUT=1
+meson setup build --cross-file=cross-file/xxxxxx.ini -Dbmd_bootloader=true
 ```
 
-or, if you are using a PCB (printed circuit board) as a shield for your Black Pill F4, run:
+2. Compile the firmware and bootloader
 
 ```sh
-cd blackmagic
-make clean
-make PROBE_HOST=blackpill-f4x1cx SHIELD=1
+ninja -C build_xxxxxx xxxxxx_firmware.bin
```

Comment:
`ninja -C build-blackpill-f411ce blackpill-f411ce-firmware.bin`
`ninja: error: unknown target 'blackpill-f411ce_firmware.bin'`

But it is OK:
`ninja -C build-blackpill-f411ce blackmagic_blackpill_f411ce_firmware.bin`

Reply:
Thanks so much for reviewing this, seems I had some copy/paste erros. Hopefully those are fixed. I think I tested all the comamdns for F411CE.

Please let me know if anything else looks wrong.